### PR TITLE
pythran: migrate to `python@3.13`

### PR DESCRIPTION
**Pull request by chenrui333**
pythran: migrate to `python@3.13`

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> /opt/homebrew/opt/pythran/bin/pythran -DUSE_XSIMD -fopenmp -march=native /private/tmp/pythran-test-20241013-14245-1xd3lk/arc_distance.py
  clang: error: unsupported option '-fopenmp'
  WARNING: Compilation error, trying hard to find its origin...
  Compilation error, trying hard to find its origin...
  WARNING: Nope, I'm going to flood you with C++ errors!
  Nope, I'm going to flood you with C++ errors!
  CRITICAL: Cover me Jack. Jack? Jaaaaack!!!!
  E: error: command '/usr/bin/clang++' failed with exit code 1
  Cover me Jack. Jack? Jaaaaack!!!!
  E: error: command '/usr/bin/clang++' failed with exit code 1
  Error: pythran: failed
  Error: pythran: failed
```

File: Formula/p/pythran.rb [hunk 1]
```diff
@@ -84,9 +89,7 @@ def arc_distance(theta_1, phi_1, theta_2, phi_2):
         return distance_matrix
     EOS
     # Test with configured gcc to detect breakages from gcc major versions and for OpenMP support
-    with_env(CC: nil, CXX: nil) do
-      system pythran, "-DUSE_XSIMD", "-fopenmp", "-march=native", testpath/"arc_distance.py"
-    end
```
Comment (carlocab) [inline Formula/p/pythran.rb:89]: We shouldn't change this.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/p/pythran.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/p/pythran.rb:89]: I see 👍

Comment (chenrui333) [reply Formula/p/pythran.rb:89]: now seeing the same in https://github.com/Homebrew/homebrew-core/pull/196198 maybe something not from pythran?

**Pull request closed without merging**